helix-term/src/commands/typed.rs

```diff
@@ -1091,20 +1091,30 @@ fn change_current_directory(
     }
 
     let dir = match args.first() {
-        Some(input_path) => {
+        Some(Cow::Borrowed("-")) => cx.editor.last_cwd.clone(),
+        Some(input_path) => Some(
             helix_stdx::path::expand_tilde(Path::new(input_path.as_ref()).to_owned())
                 .deref()
-                .to_path_buf()
-        }
-        None => home_dir()?.as_path().to_owned(),
+                .to_path_buf(),
+        ),
+        None => Some(home_dir()?.as_path().to_owned()),
     };
 
-    helix_stdx::env::set_current_working_dir(dir)?;
+    if let Some(dir) = dir {
+        helix_stdx::env::set_current_working_dir(&mut cx.editor.last_cwd, dir)?;
```

Comment:
Rather than passing this in you can set it on another line:

```rust
helix_stdx::env::set_current_working_dir(dir)?;
cx.editor.last_cwd = Some(dir);
```

Reply:
But the `last_cwd` should be set to the value of `helix_stdx::env::current_working_dir()` before the call to `set_current_working_dir` rather than the same thing as `dir`. A small refactor that would make this easier is having `set_current_working_dir` return `std::io::Result<Option<PathBuf>>` and using `cwd.replace(path)` so that the prior cwd is returned. Then it can be one line:

```rust
cx.editor.last_cwd = helix_stdx::env::set_current_working_dir(dir)?;
```